src/cbottle/training/loop.py

```diff
@@ -120,6 +121,13 @@ def _setup_datasets(self):
     def _setup_networks(self):
         self.ddp = self.net = self.get_network()
         self.net.train().requires_grad_(True).to(self.device)
+        if self.channels_last:
+            self.net = self.net.to(memory_format=torch.channels_last)
+        self.base_net = self.net
+        if self.compile:
+            # Compiling must return a new object and not modify the original net.
+            # Need a reference to the original uncompiled net so parameter names are consistent in checkpointing.
+            self.net = self.compile_network(self.net)
```

Comment:
Instead of changing the net attribute (which in theory is part of the public API and used many places), you should instead make a new internal attribute for the compiled network.

```suggestion
        self._net_forward = self.net
        if self.compile:
            # Compiling must return a new object and not modify the original net.
            # Need a reference to the original uncompiled net so parameter names are consistent in checkpointing.
            self._net_forward = self.compile_network(self.net)
```